docs/docsite/rst/community/collection_contributors/collection_integration_about.rst

```diff
@@ -0,0 +1,156 @@
+
+There are :ref:`two kinds of integration tests <collections_adding_integration_test>` used in collections:
+
+* integration tests that use Ansible roles
+* integration tests that use ``runme.sh``.
+
+This section focuses on integration tests that use Ansible roles.
+
+Integration tests check modules with playbooks that invoke those modules. The tests pass standalone parameters and their combinations, check what the module or plugin reports with the :ref:`assert <ansible_collections.ansible.builtin.assert_module>` module, and the actual state of the system after each task.
+
+Integration test example
+-------------------------
+
+Let's say we want to test the ``postgresql_user`` module invoked with the ``name`` parameter. We expect that the module will both create a user based on the provided value of the ``name`` parameter and will report that the system state has changed. We cannot rely on only what the module reports. To be sure that the user has been created, we query our database with another module to see if the user exists.
+
+.. code-block:: yaml
+
+  - name: Create PostgreSQL user and store module's output to the result variable
+    postgresql_user:
+      name: test_user
+    register: result
+
+  - name: Check the module returns what we expect
+    assert:
+      that:
+        - result is changed
+
+  - name: Check actual system state with another module, in other words, that the user exists
+    postgresql_query:
+      query: SELECT * FROM pg_authid WHERE rolename = 'test_user'
+    register: query_result
+
+  - name: We expect it returns one row, check it
+    assert:
+      that:
+        - query_result.rowcount == 1
+
+
+Details about integration tests
+--------------------------------
+
+The basic entity of an Ansible integration test is a ``target``. The target is an :ref:`Ansible role <playbooks_reuse_roles>` stored in the ``tests/integration/targets`` directory of the collection repository. The target role contains everything that is needed to test a module.
+
+The names of targets contain the module or plugin name that they test. Target names that start with ``setup_`` are usually executed as dependencies before module and plugin targets start execution. See :ref:`collection_creating_integration_tests` for details.
+
+To run integration tests, we use the ``ansible-test`` utility that is included in the ``ansible-core`` and ``ansible`` packages. See :ref:`collection_run_integration_tests` for details. After you finish your integration tests, see to :ref:`collection_quickstart` to learn how to submit a pull request.
+
+.. _collection_integration_prepare:
+
+Preparing for integration tests
+================================
+
+To prepare for developing integration tests:
+
+#. :ref:`Set up your local environment <collection_prepare_environment>`.
+
+#. Determine if integration tests already exist.
+
+  .. code-block:: bash
+
```

Comment:
To identify that the next line is a command and it is associated with the code-bash is there any need for the space in between? The indentation is not consistence with other code-blocks.

Someone might find it confusing to understand those are commands expected to be run by the user.

Reply:
The `code-block` does seem to require a blank line between it and the actual code. That said, you are correct that the indentation overall isn't consistent so I'll fix that across the files in a separate commit. Thanks for spotting that!